src/java.base/share/classes/java/io/ConsoleImpl.java

```diff
@@ -303,7 +303,6 @@ public int read(char[] cbuf, int offset, int length)
                                  */
                                 if (cbuf == rcb) {
                                     cbuf = grow();
-                                    end = cbuf.length;
```

Comment:
Initially this looked like an unintentionally change of logic to me, but when I see the current mainline code in IntelliJ, it does note that this assignment is then never used and it's right - after assigning `end` here, it never gets used and we ultimately end up with a `return` statement which is independent of this `end` value. So this change looks fine.

Reply:
In fact, that was suggested by IntelliJ 🙂